scripts/eip1559_deployer.ts

```diff
@@ -0,0 +1,449 @@
+import postageABI from '../artifacts/src/PostageStamp.sol/PostageStamp.json';
+import redisABI from '../artifacts/src/Redistribution.sol/Redistribution.json';
+import stakingABI from '../artifacts/src/Staking.sol/StakeRegistry.json';
+import oracleABI from '../artifacts/src/PriceOracle.sol/PriceOracle.json';
+import bzzTokenABI from '../artifacts/src/TestToken.sol/TestToken.json';
+
+let account: ethers.Wallet;
+
+interface DeployedContract {
+  abi: Array<unknown>;
+  bytecode: string;
+  address: string;
+  block: number;
+  url: string;
+}
+
+interface DeployedData {
+  chainId: number;
+  networkId: number;
+  contracts: {
+    postageStamp: DeployedContract;
+    redistribution: DeployedContract;
+    staking: DeployedContract;
+    priceOracle: DeployedContract;
+    bzzToken: DeployedContract;
+  };
+}
+
+interface Mnemonic {
+  mnemonic: string;
+}
+
+const blockChainVendor = hre.network.name;
+
+async function main(deployedData: DeployedData = testnetData) {
+  await setConfigurations();
+
+  switch (blockChainVendor) {
+    case 'mainnet':
+      deployedData = mainnetData;
+      break;
+    case 'testnet':
+    default:
+      deployedData = testnetData;
+      break;
+  }
+  //fill deployedData with already compiled data e.g. bytecode and ABI
+  deployedData = await writeResult(
+    await writeURL(
+      await rolesSetter(
+        await deployRedistribution(
+          await deployPriceOracle(
+            await deployPostageStamp(
+              await deployStakeRegistry(await deployBzzToken(await setCompiledData(deployedData)))
+            )
+          )
+        )
+      )
+    )
+  );
+  await verifier(deployedData);
+}
+
+async function setConfigurations() {
+  const infuraToken = process.env.INFURA_TOKEN;
+  if (
+    infuraToken === '' ||
+    (infuraToken === undefined && (blockChainVendor == 'testnet' || blockChainVendor == 'mainnet'))
+  ) {
+    throw new Error('Please set your INFURA_TOKEN in a .env file');
+  }
+  let wallet: ethers.Wallet;
+  if (Array.isArray(hre.network.config.accounts)) {
+    if (hre.network.config.accounts.length > 1) {
+      throw new Error('only 1 private key expected');
+    }
+    wallet = new ethers.Wallet(hre.network.config.accounts[0] as string);
+  } else if (isMnemonic(hre.network.config.accounts)) {
+    wallet = ethers.Wallet.fromMnemonic(hre.network.config.accounts.mnemonic);
+  } else {
+    throw new Error('unknown type');
+  }
+  switch (blockChainVendor) {
+    case 'testnet':
+      account = wallet.connect(new ethers.providers.JsonRpcProvider('https://goerli.infura.io/v3/' + infuraToken));
+      break;
+    case 'mainnet':
+      account = wallet.connect(new ethers.providers.JsonRpcProvider('https://mainnet.infura.io/v3/' + infuraToken));
+      break;
+    default:
+      account = wallet.connect(hre.ethers.provider);
+  }
+}
+
+function isMnemonic(param: unknown): param is Mnemonic {
+  return typeof param === 'object' && typeof (param as Mnemonic).mnemonic === 'string';
```

Comment:
This check doesn't make much sense since the typeof check: 'mnemonic' always has type 'string' .  Can be rewritten as: `return typeof param === 'object' && param != null && 'mnemonic' in param;`